lib/manager/selectors/index.js

```diff
@@ -1,11 +1,205 @@
+  colSpan?: string,
+  dataField?: string,
+  isKey?: boolean,
+  name: string,
+  row: string,
+  rowSpan?: string,
+  width: number
+}
+
+type Filters = {
+  showArrivals: boolean,
+  timepointFilter: boolean
+}
+
+type Stop = {
+  stop_id: string,
+  stop_lat: number,
+  stop_lon: number,
+  stop_name: string
+}
+
+type Timetables = {
+  data: {
+    feed: {
+      patterns: Array<{
+        stops: Array<Stop>,
+        trips: Array<{
+          direction_id: number,
+          pattern_id: string,
+          service_id: string,
+          stop_times: Array<{
+            arrival_time: number,
+            departure_time: number,
+            stop_id: string,
+            stop_sequence: number,
+            timepoint: ?boolean
+          }>,
+          trip_headsign: string,
+          trip_id: string,
+          trip_short_name: string
+        }>
+      }>
+    }
+  }
+}
+
+const getActiveProjectId = state => state.projects.active
+const getProjects = state => state.projects.all
 
 export const getActiveProject = createSelector(
-  [ getActiveProjectId, getProjects ],
+  [getActiveProjectId, getProjects],
   (activeProjectId, projects) => {
     return projects && projects.find(p => p.id === activeProjectId)
   }
 )
+
+export const getTimetableData: any => {
+  columns: Array<Column>,
+  rows: Array<any>
+} = createSelector(
+  [state => state.gtfs.timetables, state => state.gtfs.filter],
+  (timetables: Timetables, filters: Filters) => {
+    if (!timetables.data.feed) return {columns: [], rows: []}
+
+    // assume 1st pattern is the only one we're interested in
+    const pattern = timetables.data.feed.patterns[0]
+
+    // there could be no patterns found or no trips associate with that pattern
+    // if eiter happens, return blank
+    if (!pattern || pattern.trips.length === 0) return {columns: [], rows: []}
+
+    // create stop lookup because the stops come back out-of-order
+    const stopLookup: { [string]: Stop } = {}
+    const stopHasStopTimeWithTime: { [string]: boolean } = {}
+    pattern.stops.forEach(stop => {
+      stopLookup[stop.stop_id] = stop
+    })
+
+    // begin making column headers with trip info
+    let columns: Array<Column> = [
+      {
+        dataField: 'tripId',
+        isKey: true,
+        name: 'Trip ID',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      },
+      {
+        dataField: 'tripHeadsign',
+        name: 'Trip Headsign',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      },
+      {
+        dataField: 'tripShortName',
+        name: 'Trip Short Name',
+        rowSpan: filters.showArrivals ? '2' : '1',
+        row: '0',
+        width: 150
+      }
+    ]
+
+    // make rest of the row headers by making stop names as headers
+    // assume that the first trip is representative of stops in timetable
+    const firstTrip = pattern.trips[0]
+
+    // sort stop times by stop_sequence just in case
```

Comment:
Does the sort really need to happen here?  Doesn't gtfs-lib handle that already?

Reply:
ah, I guess it does